ui/app/app.js

```diff
@@ -299,6 +301,91 @@ App.prototype.renderDropdown = function () {
   ])
 }
 
+App.prototype.renderBackToInitButton = function () {
+  var props = this.props
+  var button = null
+  var backButton = function (style, justArrow = false) {
```

Comment:
Defining this method within a method looks really weird to me. I'd rather `backButton` be declared outside of this function scope, so this function is mostly about the logic used to make the render call, and not bundling it all together into one giant stream.